### PR TITLE
Check field type before appending

### PR DESCRIPTION
**Pull request by ryantxu**
If the input for a field changes we currently panic -- we should do something better than that

### CODE REVIEW

**Comment by leeoniya** [comment]
image

File: telemetry/telegraf/convert_test.go [hunk 1]
```diff
@@ -186,14 +214,6 @@ func TestConverter_Convert_NumFrameFields_LabelsColumn(t *testing.T) {
 	require.JSONEqf(t, string(frameJSON), string(want), "not matched with golden file")
 }
 
-func TestConverter_Convert_LabelsColumn_MixedNumberTypes_Panics(t *testing.T) {
```
Comment (ryantxu) [inline telemetry/telegraf/convert_test.go:189]: this tested explicitly that chaning the type panics -- silently replacing with null is not great, but panic seems pretty bad for user enter data

File: telemetry/telegraf/convert.go [hunk 2]
```diff
@@ -218,7 +218,17 @@ func (s *metricFrame) append(m influx.Metric) error {
 			return err
 		}
 		if index, ok := s.fieldCache[f.Key]; ok {
-			s.fields[index].Append(v)
+			field := s.fields[index]
+			if ft != field.Type() {
+				fmt.Printf("error appending value of type: %v as %v // %v\n", field.Type(), ft, v)
```
Comment (ryantxu) [inline telemetry/telegraf/convert.go:223]: @FZambia what is the best logging option here?

Comment (FZambia) [reply telemetry/telegraf/convert.go:223]: Maybe pass logger from the outside?

**Comment by ryantxu** [comment]
not necessary... this was moved to grafana core

**Pull request closed without merging**